src/ORM/Association.php

```diff
@@ -1111,6 +1112,11 @@ protected function _getClassName($alias, array $options = [])
         }
 
         $className = App::className($options['className'], 'Model/Table', 'Table') ?: 'Cake\ORM\Table';
+        if (!$className) {
+            throw new RuntimeException(
+                sprintf('Class for %s could not be found.', $options['className'])
+            );
+        }
```

Comment:
Can this error ever be raised? The `?:` should make it impossible.

Reply:
Can we make automagic tables and/or entities opt-out for 4.x through Configure::?

Reply:
Fixed

Reply:
@inoas Possibly. It could be an opt-in mode that makes debugging auto-table issues easier.